### PR TITLE
LB-230: Configure scheduler to do regular data dumps

### PR DESCRIPTION
**Pull request by paramsingh**
# Summary

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other

# Problem

* JIRA ticket (_optional_): [LB-230](https://tickets.metabrainz.org/browse/LB-230)

# Solution

Add an entry to crontab.

### CODE REVIEW

**Comment by paramsingh** [comment]
Build should be fixed after https://github.com/metabrainz/listenbrainz-server/pull/304/commits/013d3c64f66cb03be53f51c15e76453045d09cbb gets merged.

**Comment by mayhem** [comment]
@brainzbot, test this please.

**Comment by paramsingh** [comment]
Brainzbot doesn't work right now for LB, let me rebase and travis should retest this.

File: crontab [hunk 1]
```diff
@@ -1,4 +1,5 @@
 PATH=/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin
 PYTHONPATH=/usr/local/lib/python36.zip:/usr/local/lib/python3.6:/usr/local/lib/python3.6/lib-dynload:/usr/local/lib/python3.6/site-packages:/usr/local/lib/python3.6/site-packages/messybrainz-99.0.0.dev0-py3.6.egg
 GOOGLE_APPLICATION_CREDENTIALS=/code/credentials/bigquery-credentials.json
-* * * * * python /code/listenbrainz/manage.py stats calculate > /var/log/cron.log 2>&1
+* * * * * python /code/listenbrainz/manage.py stats calculate > /var/log/stats.log 2>&1
```
Comment (mayhem) [inline crontab]: Do you really want to calculate stats once a minute?

File: crontab [hunk 2]
```diff
@@ -1,4 +1,5 @@
...
+00 00 * * 0 python /code/listenbrainz/manage.py dump create > /var/log/dump_create.log 2>&1
```
Comment (mayhem) [inline crontab]: And just to be clear, do data dumps once a week?

Comment (alastair) [reply crontab]: I'm not sure if this is a MeB standard, but when I do crontabs I always comment each command saying what it is and when it runs

File: Dockerfile [hunk 3]
```diff
@@ -51,3 +51,5 @@ ADD crontab /etc/cron.d/crontab
 RUN chmod 0644 /etc/cron.d/crontab
 RUN touch /var/log/cron.log
 RUN crontab /etc/cron.d/crontab
+
+CMD service cron stop
```
Comment (mayhem) [inline Dockerfile]: This means that cron is running for no image, right? And it is a bit of a hack, methinks. I suppose we could determine if cron needs starting/stopping in this file : https://github.com/metabrainz/listenbrainz-server/blob/master/docker/beta/uwsgi/uwsgi.service (as well as the prod version of it).

Also, changes to this Dockerfile, if we keep it should be made to Dockerfile.prod

Comment (paramsingh) [reply Dockerfile]: Seemed hacky to me too. Changing the uwsgi.service script then.

>Also, changes to this Dockerfile, if we keep it should be made to Dockerfile.prod

on it.

File: Dockerfile.prod [hunk 4]
```diff
@@ -14,6 +14,7 @@ RUN apt-get update \
                        libffi-dev \
                        libssl-dev \
                        redis-tools \
+                       cron \
```
Comment (alastair) [inline Dockerfile.prod]: it looks like cron is already in the `metabrainz/python:3.6` image, I don't know if you want to add it here again explicitly to make it clear that we use it

File: docker/beta/uwsgi/uwsgi.service [hunk 5]
```diff
@@ -1,25 +1,57 @@
...
+
 if [ "${CONTAINER_NAME}" = "listenbrainz-web-beta" ]
 then
+    bring_cron_down
     exec uwsgi /etc/uwsgi/uwsgi.ini
 fi
 
 if [ "${CONTAINER_NAME}" = "listenbrainz-influx-writer-beta" ]
 then
+    bring_cron_down
```
Comment (alastair) [inline docker/beta/uwsgi/uwsgi.service]: I don't fully understand why you bring cron down in all containers. Do you see cron starting automatically in all other containers when you bring them up?
From what I understand cron won't start automatically in all containers when you start from docker, only the specified command will execute. In this case I think it makes sense to run `cron -f` in the scheduler container, but not stop it in the others.

File: Dockerfile.prod (hunk 4 above)
Comment (mayhem) [reply Dockerfile.prod]: Ok, we now understand that this is not needed.

**Comment by mayhem** [comment]
OK, we're really close! The "down" file works, but simply removing the "down" file isn't enough, we also need to do a:
  sv restart cron
In the scheduler container and cron starts. I've verified this in a production container.

File: docker/prod/uwsgi/uwsgi.service [hunk 6]
```diff
@@ -22,4 +22,11 @@ then
     exec uwsgi /etc/uwsgi/uwsgi-api-compat.ini
 fi
 
+if [ "${CONTAINER_NAME}" = "listenbrainz-scheduler-prod" ]
+then
+    rm /etc/service/cron/down
+    sv restart cron
+    exec cron -f
```
Comment (mayhem) [inline docker/prod/uwsgi/uwsgi.service]: This line is not needed. Let's remove it, then we can merge this.

**Pull request merged**